test/util/framework/zero_dep.go

```diff
@@ -0,0 +1,257 @@
+	return Must(tc.Get20240610ClientFactory())
+}
+
+func (tc *TestContext) getAzureCredentials() (azcore.TokenCredential, error) {
+	tc.contextLock.RLock()
+	if tc.azureCredentials != nil {
+		defer tc.contextLock.RUnlock()
+		return tc.azureCredentials, nil
+	}
+	tc.contextLock.RUnlock()
+
+	tc.contextLock.Lock()
+	defer tc.contextLock.Unlock()
+
+	return tc.getAzureCredentialsUnlocked()
+}
+
+func (tc *TestContext) getAzureCredentialsUnlocked() (azcore.TokenCredential, error) {
+	if tc.azureCredentials != nil {
+		return tc.azureCredentials, nil
+	}
+
+	// if we find a desire to use the zero-dep e2e testing everywhere, we can extend this credential creation to include
+	// other options for non-Azure endpoints.  It's worth remembering that the value add
+	azureCredentials, err := azidentity.NewEnvironmentCredential(nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed building environment credential: %w", err)
+	}
+	tc.azureCredentials = azureCredentials
+
+	return tc.azureCredentials, nil
+}
+
+func (tc *TestContext) GetARMResourcesClientFactory() (*armresources.ClientFactory, error) {
+	tc.contextLock.RLock()
+	if tc.clientFactory20240610 != nil {
+		defer tc.contextLock.RUnlock()
+		return tc.armResourcesClientFactory, nil
+	}
+	tc.contextLock.RUnlock()
+
+	tc.contextLock.Lock()
+	defer tc.contextLock.Unlock()
+
+	return tc.getARMResourcesClientFactoryUnlocked()
+}
+
+func (tc *TestContext) getARMResourcesClientFactoryUnlocked() (*armresources.ClientFactory, error) {
+	if tc.armResourcesClientFactory != nil {
+		return tc.armResourcesClientFactory, nil
+	}
+
+	creds, err := tc.getAzureCredentialsUnlocked()
+	if err != nil {
+		return nil, err
+	}
+	clientFactory, err := armresources.NewClientFactory(tc.subscriptionID, creds, nil)
+	if err != nil {
+		return nil, err
+	}
+	tc.armResourcesClientFactory = clientFactory
+
+	return tc.armResourcesClientFactory, nil
+}
+
+func (tc *TestContext) Get20240610ClientFactory() (*hcpapi20240610.ClientFactory, error) {
+	tc.contextLock.RLock()
+	if tc.clientFactory20240610 != nil {
+		defer tc.contextLock.RUnlock()
+		return tc.clientFactory20240610, nil
+	}
+	tc.contextLock.RUnlock()
+
+	tc.contextLock.Lock()
+	defer tc.contextLock.Unlock()
+
+	return tc.get20240610ClientFactoryUnlocked()
+}
+
+func (tc *TestContext) get20240610ClientFactoryUnlocked() (*hcpapi20240610.ClientFactory, error) {
+	if tc.clientFactory20240610 != nil {
+		return tc.clientFactory20240610, nil
+	}
+
+	creds, err := tc.getAzureCredentialsUnlocked()
+	if err != nil {
+		return nil, err
+	}
+	clientFactory, err := hcpapi20240610.NewClientFactory(tc.subscriptionID, creds, nil)
+	if err != nil {
+		return nil, err
+	}
+	tc.clientFactory20240610 = clientFactory
+
+	return tc.clientFactory20240610, nil
+}
+
+// subscriptionID returns the value of CUSTOMER_SUBSCRIPTION environment variable
+func subscriptionID() string {
+	// can't use gomega in this method since it is used outside of It()
```

Comment:
maybe I am dense, but what would you want to use gomega for in such a method ... ?

Reply:
No idea. That one was copied out of an existing setup package